cou/apps/auxiliary.py

```diff
@@ -204,6 +204,23 @@ def _get_change_require_osd_release_step(self) -> PreUpgradeStep:
 class OvnPrincipal(AuxiliaryApplication):
     """Ovn principal application class."""
 
+    def upgrade_plan_sanity_checks(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> None:
+        """Run sanity checks before generating upgrade plan.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate upgrade plan, defaults to None
+        :type units: Optional[list[Unit]], optional
+        :raises ApplicationError: When enable-version-pinning is True
+        """
+        super().upgrade_plan_sanity_checks(target, units)
+        if self.config["enable-version-pinning"].get("value"):
+            raise ApplicationError(
+                f"'{self.name}' should set 'enable-version-pinning' before upgrading"
```

Comment:
Didn't mention what value to be set for `enable-version-pinning`.
```suggestion
                f"Cannot upgrade '{self.name}'. 'enable-version-pinning' must be set to `false`."
```